hal/hal/rng_api.h

```diff
@@ -0,0 +1,74 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2016 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef MBED_RNG_API_H
+#define MBED_RNG_API_H
+
+#include <stddef.h>
+#include "device.h"
+
+#if DEVICE_RNG
+
```

Comment:
To avoid user application software taking all the entropy, we should make it clear in some comment that this API is for the exclusive use of mbed TLS when present.

Ideally, we should have some lock to restrict it to mbed TLS only. (Although I'm not sure how we could do that without cost).